jaxdecomp/_src/fft.py

```diff
@@ -75,46 +81,38 @@ def sfft_abstract_eval(x, fft_type, pdims, global_shape, adjoint):
     axis = 2
 
   output_shape = None
+  match fft_type:
+    case xla_client.FftType.FFT:
+      # FFT is X to Y to Z so Z-Pencil is returned
+      # Except if we are doing a YZ slab in which case we return a Y-Pencil
+      transpose_shape = (1, 2, 0) if not yz_slab else (2, 0, 1)
+      transposed_pdims = pdims if not yz_slab else pdims[::-1]
```

Comment:
Minor comment : I used to love these concise one liners but nowadays I personally find them less readable in terms of code logic, especially since you are using a negation.

```python
if yz_slab:
    # in the specific case of a YZ slab we return a Y-Pencil
    transpose_shape = (2, 0, 1)
    transposed_pdims = pdims[::-1]
else:
    # otherwise FFT is X to Y to Z so Z-Pencil is returned
    transpose_shape = (1, 2, 0)
    transposed_pdims = pdims
```